src/Symfony/Component/DependencyInjection/Compiler/ValidateEnvPlaceholdersPass.php

```diff
@@ -51,13 +51,22 @@ public function process(ContainerBuilder $container)
         foreach ($resolvingBag->getEnvPlaceholders() + $resolvingBag->getUnusedEnvPlaceholders() as $env => $placeholders) {
             $values = [];
             if (false === $i = strpos($env, ':')) {
-                $default = $defaultBag->has("env($env)") ? $defaultBag->get("env($env)") : self::TYPE_FIXTURES['string'];
-                $defaultType = null !== $default ? get_debug_type($default) : 'string';
+                [$default, $defaultType] = $this->getParameterDefaultAndDefaultType("env($env)", $defaultBag);
                 $values[$defaultType] = $default;
             } else {
                 $prefix = substr($env, 0, $i);
-                foreach ($envTypes[$prefix] ?? ['string'] as $type) {
-                    $values[$type] = self::TYPE_FIXTURES[$type] ?? null;
+
+                if ('default' === $prefix) {
+                    [$parameter, $envVar] = explode(':', substr($env, $i + 1));
```

Comment:
there can be more than just an env var after the parameter:
`env(default:foo:json:BAR)`